### PR TITLE
[5.1] Replace Presenters by Casters for PsySH 0.5

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Once bobthecow/psysh#184 is merged and PsySH 0.5 is released, presenters should be replaced by casters.
This PR ports the existing Laravel presenters to casters.
To be honest, I didn't test the code so if someone wants to do so before me, she or he can :-)

### CODE REVIEW

File: src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        'databasePath',
+        'langPath',
+        'publicPath',
+        'storagePath',
+    ];
+
+    /**
+     * Get an array representing the properties of an Application object.
+     *
+     * @param  Application  $value
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php]: Please use fqcns

File: src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+     * @param  array  $a
+     * @param  Stub  $stub
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php]: same

File: src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        return [
+            Caster::PREFIX_VIRTUAL.'all' => $coll->all(),
+        ];
+    }
+
+    /**
+     * Get an array representing the properties of a Model object.
+     *
+     * @param  Application  $value
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php]: etc :)

**Comment by GrahamCampbell** [comment]
Looks good. What's the time scale on symfony 2.7.2 and psysh 0.5.0?

**Comment by GrahamCampbell** [comment]
I don't think we should merge this until both are tagged.

**Comment by nicolas-grekas** [comment]
I agree, we should wait for them to be tagged.
The schedule is unspecified but I'd say no more than a few weeks.

**Comment by GrahamCampbell** [comment]
:)

File: src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     * @return array
+     */
+    public static function castApplication(Application $app, array $a, Stub $stub, $isNested, $filter = 0)
+    {
+        $a = [];
+
+        foreach (self::$appProperties as $property) {
+            try {
+                $val = $value->$property();
+                if ( ! is_null($val)) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/Tinker/IlluminateCaster.php]: no spaces either side of ! please

**Comment by taylorotwell** [comment]
Can someone ping me when 0.5 is actually released then I will come back to it. Just going to close it for no so it's not in my PR work list.

**Pull request closed without merging**